pkg/statedb/reconciler/reconciler.go

```diff
@@ -107,12 +107,16 @@ func WaitForReconciliation[Obj any](ctx context.Context, db *statedb.DB, table s
 }
 
 func (r *reconciler[Obj]) loop(ctx context.Context, health cell.HealthReporter) error {
+
 	var fullReconTickerChan <-chan time.Time
 	if r.Config.FullReconcilationInterval > 0 {
 		fullReconTicker := time.NewTicker(r.Config.FullReconcilationInterval)
 		defer fullReconTicker.Stop()
 		fullReconTickerChan = fullReconTicker.C
 	}
+	if r.Config.RateLimiter != nil {
+		defer r.Config.RateLimiter.Stop()
+	}
```

Comment:
Nit. May be worth to move it to a separate commit, as it appears an unrelated fix.